File: Java/src/main/java/QuestionBank/Solution001_100/Solution82.java
package QuestionBank.Solution001_100;

import tool.ListNode;

/**
 * @author windows
 */
public class Solution82 {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null){
            return null;
        }
        ListNode dummy = new ListNode(-101, head);
        ListNode cur = dummy;
        while (cur.next != null && cur.next.next != null){
            if (cur.next.val == cur.next.next.val){
                int x = cur.next.val;
                while(cur.next != null && cur.next.val == x){
                    cur.next = cur.next.next;
                }
            }else {
                cur = cur.next;
            }
        }
        return dummy.next;
    }


    public static void main(String[] args) {
        Solution82 solution82 = new Solution82();
    }
}

File: C++/src/Solution1_100/Solution82.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution82.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

//Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

ListNode* deleteDuplicates(ListNode* head) {
    ListNode* dummy = new ListNode(0, head);
    ListNode* cur = dummy;
    while (cur->next && cur->next->next){
        if (cur->next->val == cur->next->next->val){
            int x = cur->next->next->val;
            while (cur->next && cur->next->val == x){
                cur->next = cur->next->next;
            }
        }else{
            cur = cur->next;
        }
    }
    return dummy->next;
}